src/common/exif.cc

```diff
@@ -1110,6 +1120,16 @@ int dt_exif_write_blob(uint8_t *blob, uint32_t size, const char *path, const int
       dt_remove_exif_keys(imgExifData, keys, n_keys);
     }
 
+    // remove subimage* trees, related to thumbnails or HDR usually
+    for(Exiv2::ExifData::iterator i = imgExifData.begin(); i != imgExifData.end();)
+    {
+      std::string needle = "Exif.SubImage";
+      if(i->key().compare(0, needle.length(), needle) == 0)
```

Comment:
I'm not sure about which function you are talking about.
`dt_exif_read_blob()` is **only** called to get the exif blob to immediately write it into output image via `dt_exif_write_blob()`.
So the blob that `dt_exif_read_blob()` returns is never actually used by dt.